src/psfmachine/machine.py

```diff
@@ -617,21 +633,45 @@ def _time_bin(self, npoints=200):
             The binned flux, raw
         flux_binned: np.ndarray
             The binned flux, whitened by the mean of the flux in time
-        flux_err_binned:
+        flux_err_binned:: np.ndarray
             The binned flux error, whitened by the mean of the flux
+        pc1_smooth and pc2_smooth: np.ndarray
+            Smooth version of the median poscorr vectors 1 and 2 used for time correction.
+        pc1_bin and pc2_bin: np.ndarray
+            Binned version of poscorrs vectors 1 and 2 used for setting knots.
         """
 
-        # Where there are break points in the data
+        # Where there are break points in the time array
         splits = np.append(
-            np.append(0, np.where(np.diff(self.time) > 0.1)[0]), len(self.time)
+            np.append(0, np.where(np.diff(self.time) > 0.1)[0] + 1), len(self.time)
         )
+        # if using poscorr, find and add discontinuity in poscorr data
+        if hasattr(self, "pos_corr1") and self.time_corrector == "pos_corr":
+            # take the scene-median poscorr
+            mpc1 = np.nanmedian(self.pos_corr1, axis=0)
+            mpc2 = np.nanmedian(self.pos_corr2, axis=0)
+
+            # find poscorr discontinuity in each axis
+            grads1 = np.gradient(mpc1, self.time)
+            grads2 = np.gradient(mpc2, self.time)
+            splits1 = np.where(grads1 > 7 * grads1.std())[0]
+            splits2 = np.where(grads2 > 7 * grads2.std())[0]
+            # merging breaks
+            splits = np.unique(np.concatenate([splits, splits1[1::2], splits2[1::2]]))
+            del grads1, grads2, splits1, splits2
+
```

Comment:
I don't understand what `splits_a` and `_b` is, and I don't know why there is a +100 here

Reply:
this is a way to ensure no knots fall in the poscorr/time discontinuities